src/dps.c

```diff
@@ -1166,10 +1175,66 @@ static void NodeRun(void* arg)
     }
 }
 
+static DPS_Status SetCurve(DPS_KeyStoreRequest* request, const DPS_Key* key)
+{
+    int8_t* curve = request->data;
+    uint8_t d[EC_MAX_COORD_LEN];
+
+    switch (key->type) {
+    case DPS_KEY_EC:
+        *curve = key->ec.curve;
+        return DPS_OK;
+    case DPS_KEY_EC_CERT:
+        if (key->cert.privateKey) {
+            return ParsePrivateKey_ECDSA(key->cert.privateKey, key->cert.privateKeyLen,
+                                         key->cert.password, key->cert.passwordLen,
+                                         curve, d);
+        }
+        /* FALLTHROUGH */
+    default:
+        return DPS_ERR_MISSING;
+    }
+}
+
+static DPS_Status GetSignatureAlgorithm(DPS_KeyStore* keyStore, const uint8_t* id, size_t idLen, int8_t* alg)
+{
+    DPS_KeyStoreRequest request;
+    int8_t curve = 0;
```

Comment:
Why a signed value?

Reply:
COSE defines the curve type as signed (although all the negative values are reserved).  I'll see if I can use the same enum at both COSE and DPS layers.